rpcs3/rpcs3qt/memory_viewer_panel.cpp

```diff
@@ -589,6 +590,14 @@ memory_viewer_panel::memory_viewer_panel(QWidget* parent, std::shared_ptr<CPUDis
 
 		idm::remove_verify<memory_viewer_handle>(id, handle_ptr);
 	});
+
+	if (!g_fxo->try_get<memory_viewer_fxo>())
+	{
+		g_fxo->init<memory_viewer_fxo>();
+	}
+
+	auto& fxo = g_fxo->get<memory_viewer_fxo>();
+	fxo.last_opened = this;
```

Comment:
You probably should set this to nullptr with something like this:
```
connect(this, destroyed, this, [this]()
{
	if (auto fxo = g_fxo->try_get<memory_viewer_fxo>())
	{
		if (this == fxo->last_opened)
			fxo->last_opened = nullptr;
	}
});
```

Reply:
why is it done in the constructor and not at ShowAtPc? there are other memory viewer ctors in the code.

Reply:
> why is it done in the constructor and not at ShowAtPc? there are other memory viewer ctors in the code.

I'm not sure I understand. In what situation would a `memory_viewer_panel` be initialized in another constructor?

Reply:
No I mean, there are other places memory_viewer_panel is used. They may differ in type such as for SPU etc, we do not want them be connected to ShowAtPc method.

Reply:
bump

Reply:
> bump

Updated. It will now open / cleanup the proper memory viewer type based on the context.